gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnAutoScalingManager.java

```diff
@@ -299,27 +325,65 @@ void runInternal() {
           // Remove this instance if existed in the tracking map.
           instanceIdleSince.remove(participant);
         }
+
+        if(instancesInInitState.contains(participant)) {
+          instanceInitStateSince.putIfAbsent(participant, System.currentTimeMillis());
+          if (!isInstanceStuckInInitState(participant)) {
+            // release the corresponding container as the helix task is stuck in INIT state for a long time
+            log.info("Instance {} is stuck in INIT state for a long time, releasing the container", participant);
+            // get containerInfo of the helix participant
+            YarnService.ContainerInfo containerInfo = yarnService.getContainerInfoGivenHelixParticipant(participant);
+            if(containerInfo != null) {
+              containersToRelease.add(containerInfo.getContainer());
+              instanceInitStateSince.remove(participant);
+              inUseInstances.remove(participant);
+            } else {
+              log.warn("ContainerInfo for participant {} is not found", participant);
```

Comment:
If the participant info is not found, shall we remove it from `instanceInitStateSince` as well? When this will get removed from `instanceInitStateSince` for this case ?